rts/Sim/Units/Unit.cpp

```diff
@@ -1015,7 +1015,7 @@ void CUnit::SlowUpdate()
 	}
 
 	// FIXME: tidal part should be under "if (activated)"?
-	AddEnergy((unitDef->energyMake + unitDef->tidalGenerator * envResHandler.GetCurrentTidalStrength()) * 0.5f);
+	AddEnergy((unitDef->tidalGenerator * envResHandler.GetCurrentTidalStrength()) * 0.5f);
```

Comment:
Ah I see, the above was probably accounted here.

Reply:
Yes. This is still a logic change because a unit can now use both metal and energy produced to pay its own upkeep (previously it could with metal but not with energy, because metal was produced before upkeep but energy was produced after). I don't think any game is affected though.